### PR TITLE
Refman consistency check

### PR DESCRIPTION
**Pull request by wkolowski**
**Kind:** documentation

This is a global consistency check for the Reference Manual. Note that this is based on PR #8135, so a lot of stuff appearing here is not mine. The important part is only the commit c6c75e1. (I hope I did everything right...)

The spelling changes were basically these:
1. implementor -> implementer
2. automatiz-, automatis- -> automat- (i.e. automatization -> automation, automatise -> automate etc.)
3. non zero, non-zero -> nonzero
4. non-associative, non associative -> nonassociative
5. file name -> filename
6. behaviour -> behavior
7. name space -> namespace
8. an hypothesis -> a hypothesis
9. semi-ring -> semiring
10. semi-field -> semifield
11. semi-decision -> semidecision
12. type class, type-class -> typeclass (I chose "typeclass" because it is written this way in commands and tactics)
13. type-check, typecheck -> type check (this seems to be the most popular variant)
14. type-inference -> type inference

Terminology: disequality now means <> and inequality means < <= > >= everywhere in the manual. I removed the word "inequation" from all places.

### CODE REVIEW

File: doc/sphinx/proof-engine/tactics.rst [hunk 1]
```diff
@@ -3731,7 +3729,7 @@ Setting implicit automation tactics
    In this case the tactic command typed by the user is equivalent to
    ``tactic``:sub:`1` ``;tactic``.
 
-   See also: ``Proof.`` in :ref:`proof-editing-mode`.
+   .. seealso:: ``Proof`` in :ref:`proof-editing-mode`.
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/tactics.rst]: ```:cmd:`Proof` ``` would be even better

File: doc/sphinx/addendum/type-classes.rst [hunk 2]
```diff
@@ -431,7 +431,7 @@ Typeclasses Transparent, Typclasses Opaque
 
 .. cmd:: Typeclasses Transparent {+ @ident}
 
-   This command defines makes the identifiers transparent during type class
+   This command defines makes the identifiers transparent during typeclass
```
Comment (Zimmi48) [inline doc/sphinx/addendum/type-classes.rst]: This is independent of your changes but "defines" should be removed here.

File: doc/sphinx/proof-engine/tactics.rst [hunk 3]
```diff
@@ -4366,8 +4365,10 @@ printed with the Print Fields command.
 
       intros; field.
 
-See also: file plugins/setoid_ring/RealField.v for an example of instantiation,
-theory theories/Reals for many examples of use of field.
+.. seealso::
+
+   File plugins/setoid_ring/RealField.v for an example of instantiation,
+   theory theories/Reals for many examples of use of field.
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/tactics.rst]: These filenames could use some verbatim.

File: doc/sphinx/proof-engine/vernacular-commands.rst [hunk 4]
```diff
@@ -1130,8 +1131,10 @@ described first.
 
       There is no constant referred by :n:`@qualid` in the environment.
 
-   See also: sections :ref:`performingcomputations`,
-   :ref:`tactics-automating`, :ref:`proof-editing-mode`
+      .. seealso:
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/vernacular-commands.rst]: Missing semi-colon here as well.

File: doc/sphinx/proof-engine/vernacular-commands.rst [hunk 5]
```diff
@@ -1195,7 +1198,7 @@ described first.
    nothing prevents the user to also perform a
    ``Ltac`` `ident` ``:=`` `convtactic`.
 
-   See also: sections :ref:`performingcomputations`
+   .. seealso: :ref:`performingcomputations`
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/vernacular-commands.rst]: Same here.

**Review by Zimmi48 — COMMENTED** [review]
Sorry for the delay in doing this review. I was waiting for the integration of #8135 to avoid any spurious rebase. I hope you don't mind but I've done the rebase for you.

In my comments, only the ones about missing colons must be addressed before merging. The rest are completely optional.

File: doc/sphinx/proof-engine/tactics.rst [hunk 6]
```diff
@@ -3293,10 +3291,10 @@ command.
   Performs all the rewriting in the clause :n:`@clause`. The clause argument
   must not contain any ``type of`` nor ``value of``.
 
-See also: :ref:`Hint-Rewrite <hintrewrite>` for feeding the database of lemmas used by
-:tacn:`autorewrite`.
+.. seealso:
```
[the remark this answers is not in the document]
Comment (Zimmi48) [reply doc/sphinx/proof-engine/tactics.rst]: Oh yes :confused: Let me fix this everywhere.

**Comment by Zeimer** [comment]
@Zimmi48 

Fixed everything except the filenames - I'm not quite sure what you meant by "verbatim".

**Review by Zimmi48 — APPROVED** [review]
Let's keep it like this.

**Pull request merged**